resources/user.rb

```diff
@@ -0,0 +1,161 @@
+property :username, String, name_property: true
+property :password, String, sensitive: true, required: true
+property :first_name, String, default: ''
+property :last_name, String, default: ''
+property :email, String, default: ''
+property :roles, Array, default: []
+property :api_endpoint, String, identity: true, default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, identity: true, default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, identity: true, sensitive: true, default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  apiclient = ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+
+  begin
+    res = apiclient.run_script('get_user', username)
+    current_value_does_not_exist! if res == 'null'
```

Comment:
OPT: You could check for `.nil?` after JSON parsing

Reply:
Done